From 68e7b8b6643a4c50a8527610dbf5a7ab36cc8f24 Mon Sep 17 00:00:00 2001
From: rguenth <rguenth@138bc75d-0d04-0410-961f-82ee72b054a4>
Date: Mon, 18 Apr 2011 14:25:57 +0000
Subject: [PATCH] 2011-04-18  Richard Guenther  <rguenther@suse.de>

	PR middle-end/46364
	* g++.dg/torture/pr46364.C: New testcase.


git-svn-id: svn+ssh://gcc.gnu.org/svn/gcc/branches/gcc-4_6-branch@172650 138bc75d-0d04-0410-961f-82ee72b054a4

new file mode 100644
index 0000000..8098991
--- /dev/null
+++ b/gcc/testsuite/g++.dg/torture/pr46364.C
@@ -0,0 +1,20 @@
+// { dg-do compile }
+#include <string>
+
+void a() throw (int);
+void b(std::string const &);
+
+void c(std::string *e)
+{
+  b("");
+
+  try {
+      a();
+  } catch (...) {
+      *e = "";
+  }
+}
+
+void d() {
+    c(0);
+}
-- 
1.7.0.4

